src/devices/GoPiGo3/Movements/Vehicle.cs

```diff
@@ -0,0 +1,276 @@
+        /// You need to do some math to have the actual vehicule turning fully at 360. It depends of the reduction used.
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="degrees">degrees to turn each motor</param>
+        public void TurnRight(int speed, int degrees)
+        {
+            RunMotorSyncDegrees(new MotorPort[2] { PortLeft, PortRight }, new int[2] { speed * _correctedDir, -speed * _correctedDir }, new int[2] { degrees, degrees });
+        }
+
+        /// <summary>
+        /// Turn the vehicule left for a number of milliseconds
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="timeout">number of milliseconds to run the motors</param>
+        public void TrunLeftTime(int speed, int timeout)
+        {
+            RunMotorSyncTime(new MotorPort[2] { PortLeft, PortRight }, new int[2] { -speed * _correctedDir, speed * _correctedDir }, timeout);
+        }
+
+        /// <summary>
+        /// Turn the vehicule right for a number of milliseconds
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="timeout">number of milliseconds to run the motors</param>
+        public void TrunRightTime(int speed, int timeout)
+        {
+            RunMotorSyncTime(new MotorPort[2] { PortLeft, PortRight }, new int[2] { speed * _correctedDir, -speed * _correctedDir }, timeout);
+        }
+
+        /// <summary>
+        /// Stop the vehicule
+        /// </summary>
+        public void Stop()
+        {
+            StopMotor(PortLeft);
+            StopMotor(PortRight);
+        }
+
+        /// <summary>
+        /// Run backward for the specified number of milliseconds
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="timeout">>number of milliseconds to run the motors</param>
+        public void Backward(int speed, int timeout)
+        {
+            RunMotorSyncTime(new MotorPort[2] { PortLeft, PortRight }, new int[2] { speed * _correctedDir, speed * _correctedDir }, timeout);
+        }
+
+        /// <summary>
+        /// Run forward for the specified number of milliseconds
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="timeout">>number of milliseconds to run the motors</param>
+        public void Foreward(int speed, int timeout)
+        {
+            Backward(-speed, timeout);
+        }
+
+        /// <summary>
+        /// Return the MotorPort of the left motor 
+        /// </summary>
+        public MotorPort PortLeft { get; }
+
+        /// <summary>
+        /// Return the MotorPort of the right motor
+        /// </summary>
+        public MotorPort PortRight { get; }
+
+        /// <summary>
+        /// Is the vehicule has inverted direction, then true
```

Comment:
vehicle

Reply:
btw. InvertedDirection as is in the description sounds more descriptive as a name :smile:

Reply:
agree :-) renamed